### DISCUSSION TITLE
Liquidsoap 2.2.0 install problem with OPAM

### DISCUSSION THREAD

**Discussion by SkillenUK** [savonet/liquidsoap#3294] [Installation]
I'd be grateful if someone could offer some advice. I'm trying to install Liquidsoap 2.2.0 with OPAM, but ffmpeg-avcodec.1.1.8 fails to install. I'm running...

```
opam depext taglib mad lame cry ffmpeg ssl liquidsoap
opam install taglib mad lame cry ffmpeg ssl liquidsoap
```

During the install I see the message...

`[ERROR] The compilation of ffmpeg-avcodec.1.1.8 failed at "dune build -p ffmpeg-avcodec -j 1 @install".`

If I try to then try to install ffmpeg-avcodec.1.1.8 seperately I get...

`Requested 'libavcodec >= 58.87.100' but version of libavcodec is 58.54.100`

I'm a bit stuck as to what I need to try now. I started with an OPAM switch for ocaml 5.0.0, so it was a clean start rather than an upgrade from Liquidsoap 2.1.4. I know that the build method has changed so I have posted in discussion because I suspect it's something that I'm doing wrong rather than an issue.

Thanks,

SkillenUK.

**Comment by vitoyucepi** [comment]
Hi @SkillenUK,
- *liquidsoap=2.2.0* requires *ffmpeg=1.1.8*
- *ffmpeg=1.1.8* requires *libavcodec>=58.87.100*
- *ubuntu:20.04* provides *58.54.100*

If I were you, I would consider one of these options
1. Use liquidsoap in a container.
2. Upgrade ubuntu to 22.04.
3. Install ffmpeg from ppa, for example https://launchpad.net/~savoury1.
4. Build ffmpeg from sources.

**Reply by SkillenUK** [reply]
Thanks, it might be time to move to Ubuntu 22.04 :)
